Sources/Bow/Optics/Fold.swift

```diff
@@ -4,14 +4,58 @@ public class ForFold {}
 public typealias FoldOf<S, A> = Kind2<ForFold, S, A>
 public typealias FoldPartial<S> = Kind<ForFold, S>
 
-public class Fold<S, A> : FoldOf<S, A> {
+open class Fold<S, A> : FoldOf<S, A> {
```

Comment:
All optics can be `protocols`. I missed that in the previous review.

Reply:
The problem we have is that in Swift we cannot make those protocols be higher kinds, and the only way to implement Fold and Traversal was through extension, because of the way Swift handles generics and protocols with associated types.